probatus/feature_elimination/feature_elimination.py

```diff
@@ -526,12 +523,9 @@ def fit(
             # Get current dataset info
             current_features_set = remaining_features
             if columns_to_keep is None:
-                # Keeps the original order, while removing duplicate elements
-                remaining_removeable_features = pd.Series(current_features_set).unique()
+                remaining_removeable_features = list(set(current_features_set))
```

Comment:
Set is by definition unordered. Using such implementation for classifiers (like Lightgbm) where the order of column input is important to determine which are kept and which are not, leads to bugs. Therefore we use pd.series.unique() which guarantees the order is the same as the input minus the features which are duplicate.

Reply:
This is odd. The diff between my PR and main shown here is not right. I didn't make many of these changes. I think I know the issue. I think I based off an older version. I might have to do a fresh PR...